### PR TITLE
Add missing device types solar pv, battery storage, heat pump

### PR DESCRIPTION
**Pull request by jamesharrow**
Autogenerated from Alchemy, rename the clusters to remove 'Matter' which is consistent with #36125 

Used: https://github.com/project-chip/alchemy/releases/download/v0.8.3/alchemy-0.8.3-linux-amd64.tar.gz

`./alchemy zap --attribute="in-progress"  --sdkRoot=./connectedhomeip/ --specRoot=./connectedhomeip_spec
./connectedhomeip_spec/src/device_types/HeatPump.adoc`

`./alchemy zap --attribute="in-progress"  --sdkRoot=./connectedhomeip/ --specRoot=./connectedhomeip_spec ./connectedhomeip_spec/src/device_types/SolarPower.adoc`

`./alchemy zap --attribute="in-progress"  --sdkRoot=./connectedhomeip/ --specRoot=./connectedhomeip_spec ./connectedhomeip_spec/src/device_types/BatteryStorage.adoc`

Then
`./scripts/tools/zap_regen_all.py `

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -2698,4 +2698,76 @@ limitations under the License.
...
     </deviceType>
+    <deviceType>
+        <name>MA-heatpump</name>
+        <domain>CHIP</domain>
+        <typeName>Heat Pump</typeName>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x0309</deviceId>
+        <class>Simple</class>
+        <scope>Endpoint</scope>
+        <clusters>
```
Comment (soares-sergio) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2709]: I'm still learning how to map the spec to the code but It looks like there are a few mandatory clusters missing, according to: https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/device_types/HeatPump.adoc#device-type-requirements

ElectricalSensor and DEM at least, I believe.

Comment (soares-sergio) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2709]: Nevermind, I was looking at the wrong list. This one looks right: https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/device_types/HeatPump.adoc#cluster-requirements

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -2698,4 +2698,76 @@ limitations under the License.
...
+            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
+                <requireAttribute>CLIENT_LIST</requireAttribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2711]: Why are all these requireAttribute for mandatory attributes here?  looks like just noise...

Comment (jamesharrow) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2711]: Similar comment was raised on #36125 - it looks like this is something some clusters have and Alchemy has included.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 1 above)
Comment (jamesharrow) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2709]: Correct - the device type stuff can be confusing - the Electrical Sensor and DEM are also device types (with their own cluster requirements) that need to be included if this device type is included.

**Review by soares-sergio — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
Comment (jamesharrow) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2754]: @bzbarsky-apple @hasty is there a decision on what to do here? Are we okay to leave these in or should I manually strip them out.

I notice that #36125 has merged already which presumably has these left in?

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Approving, I guess, but I am not a fan of making XML updates harder to review.

**Pull request merged**